From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 25 Sep 2017 22:49:55 +0100
Subject: ip6_fib: Avoid ABI change in 4.9.51
Forwarded: not-needed

Commit c5cff8561d2d "ipv6: add rcu grace period before freeing
fib6_node" added an rcu_head field to the end of struct fib6_node.
As this structure is always allocated and freed inside of the ipv6
module (or built-in code), we can safely hide it from genksyms.

---
--- a/include/net/ip6_fib.h
+++ b/include/net/ip6_fib.h
@@ -68,7 +68,9 @@ struct fib6_node {
 	__u16			fn_flags;
 	int			fn_sernum;
 	struct rt6_info		*rr_ptr;
+#ifndef __GENKSYMS__
 	struct rcu_head		rcu;
+#endif
 };
 
 #ifndef CONFIG_IPV6_SUBTREES
